### DISCUSSION TITLE
Fastapi Method not allowed 405 error in Api

### DISCUSSION THREAD

**Discussion by aimahusnain** [vercel/community#7055] [Help]
### Summary

Hi,
 I'm encountering a 405 Method Not Allowed error while working on a project using Next.js, TypeScript, Tailwind CSS, and FastAPI. My project is a website for background removal, using the rembg Python package. Locally, everything works perfectly, but once deployed, the website gives the 405 error.

Here is my [deployed website](https://removebg-delta.vercel.app/) on vercel

I'm new to FastAPI and have been learning through YouTube tutorials and GitHub exploration. However, I haven't found many projects to study from. My approach is to learn by reviewing others' code to understand their methods.

The issue arises when I try to upload a file smaller than 4MB on the deployed website. Locally, there's no problem, but the deployed site returns a 405 Method Not Allowed error.

Does anyone have any suggestions on how to resolve this? Here's the link to my API GitHub repository: [GitHub Repo](https://github.com/aimahusnain/Background-Removing/blob/main/api/index.py).

### Steps to Reproduce

![2024-05-31_16-33](https://github.com/vercel/community/assets/103668742/cec47d51-8608-4c84-9e59-fda5a8f71a56)

**Comment by versecafe** [comment]
@aimahusnain can you check what the X-Vercel-Error is, I'm wondering if this is Vercel failling to allow it or fastAPI saying it's a disallowed method

**Comment by aimahusnain** [comment]
Is it possible to connect Python to Next.js instead of FastAPI?

**Reply by versecafe** [reply]
you could do a python in WASIX (Extended WASM) runtime inside Next or really any JS framework, wasmer.io is one way to go about it if you'd like to avoid python runtimes on Vercel
